### PR TITLE
Makes dash actions and Bluespace Prophecy trauma utilise teleport code instead of forced movement code.

### PR DESCRIPTION
**Pull request by Timberpoes**
## About The Pull Request

Does exactly what it says on the tin.

Dash actions such as ninja sword and hiero club now utilise bubble messages for feedback and do_teleport so they respect the various teleportation blocking flags we place on restricted/admin/weird areas.

Bluespace Prophecy teleportation also uses do_teleport for the same reasons.

## Why It's Good For The Game

Stops weird anomalies like people breaking out of places they shouldn't be allowed to and breaking in to places they shouldn't be allowed to. Protects areas on the CentCom z-level a bit better when players get sent there for whatever reasons.

## Changelog

:cl:
fix: Items with dash actions and the Bluespace Prophecy trauma now utilise proper teleportation code, disallowing teleportation to and from teleport restricted areas.
/:cl:

### CODE REVIEW

**Comment by Iplayforss13** [comment]
Whats wrong with somehow escaping centcom prison, its called an epic gamer move

File: code/modules/mining/lavaland/megafauna_loot.dm [hunk 1]
```diff
@@ -16,14 +16,14 @@
...
+/datum/action/innate/dash/hierophant/teleport(mob/user, atom/target)
 	var/dist = get_dist(user, target)
 	if(dist > HIEROPHANT_BLINK_RANGE)
-		to_chat(user, span_hierophant_warning("Blink destination out of range."))
+		user.balloon_alert(user, "Blink destination out of range.")
 		return
 	var/turf/target_turf = get_turf(target)
 	if(target_turf.is_blocked_turf_ignore_climbable())
-		to_chat(user, span_hierophant_warning("Blink destination blocked."))
+		user.balloon_alert(user, "blink destination blocked.")
```
Comment (Mothblocks) [inline code/modules/mining/lavaland/megafauna_loot.dm:26]:
```suggestion
		user.balloon_alert(user, "destination blocked!")
```

File: code/modules/mining/lavaland/megafauna_loot.dm [hunk 2]
```diff
@@ -16,14 +16,14 @@
 	// It's a simple purple beam, works well enough for the purple hiero effects.
 	beam_effect = "plasmabeam"
 
-/datum/action/innate/dash/hierophant/Teleport(mob/user, atom/target)
+/datum/action/innate/dash/hierophant/teleport(mob/user, atom/target)
 	var/dist = get_dist(user, target)
 	if(dist > HIEROPHANT_BLINK_RANGE)
-		to_chat(user, span_hierophant_warning("Blink destination out of range."))
+		user.balloon_alert(user, "Blink destination out of range.")
```
Comment (Mothblocks) [inline code/modules/mining/lavaland/megafauna_loot.dm:22]:
```suggestion
		user.balloon_alert(user, "destination out of range!")
```

The "!" denotes a failure.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Some minor balloon alert quips. 

Specifically, should be all lowercase (I think you know this, you just missed it), but also that exclamation points denote failure (which I also think you knew, going off of one used higher)

**Comment by moocowswag** [comment]
Won`t this also make it so the affected abilities can no longer function in the luxury shuttle?

This could severely negatively impact ninja and (apparently cultist / heretic?) In circumstances where they need/want to fight inside of the shuttle.

**Comment by Timberpoes** [comment]
> 
> 
> Won`t this also make it so the affected abilities can no longer function in the luxury shuttle?
> 
> This could severely negatively impact ninja (and apparently cultist / heretic cause youre changing some cult dagger items?) In circumstances where they need/want to fight inside of the shuttle. (mostly demoralizes ninja from trying to play hijack and cult/heretic from getting more converts/sacrifices from crew on shuttle before it launches)
> 
> Forgive me if im wrong I just believe this could be a gameplay altering side-affect of this PR.

There are definitely gameplay altering side effects. Areas that cannot be teleported to/from includes secret away missions (Very Good); CentCom areas (Very Good); Supply (Indifferent), Luxury (Indifferent), Medieval (Good, teleporting breaks things on it) and Arena (Indifferent) shuttles, Office in Hellfactory Ruin (Indifferent/Good harder to reach ruins), Syndie Forgottenships (Indifferent/Good harder to reach ruins) and Hibert's Hotel (Good, pocket dimension).

**Pull request merged**